Samples/SnowPal-Win-101/snowpal/SnowpalGame.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace snowpal
+{
+    public class SnowpalGame
+    {
+        private readonly List<string> _wordList = new List<string> { "apple", "banana", "cherry", "date", "fig", "grape" };
+        private const int MaxIncorrectGuessesValue = 6;
+        private static readonly Random Random = new Random();
+
+        public string CurrentWord { get; private set; }
+        public char[] GuessedWord { get; private set; }
+        public int IncorrectGuesses { get; private set; }
+        public int MaxIncorrectGuesses => MaxIncorrectGuessesValue;
+
+        public SnowpalGame()
+        {
+            StartNewGame();
+        }
+
+        public void StartNewGame()
+        {
+            var random = new Random();
+            CurrentWord = _wordList[random.Next(_wordList.Count)];
+            GuessedWord = new string('_', CurrentWord.Length).ToCharArray();
+            IncorrectGuesses = 0;
+        }
+
+        public bool GuessLetter(char letter)
+        {
+            bool isCorrect = false;
+
+            if (CurrentWord.Contains(letter))
+            {
+                for (int i = 0; i < CurrentWord.Length; i++)
+                {
+                    if (CurrentWord[i] == letter)
+                    {
+                        GuessedWord[i] = letter;
+                        isCorrect = true;
+                    }
+                }
+            }
+            else
+            {
+                IncorrectGuesses++;
+            }
```

Comment:
```suggestion
            for (int i = 0; i < CurrentWord.Length; i++)
            {
                if (CurrentWord[i] == letter)
                {
                    GuessedWord[i] = letter;
                    isCorrect = true;
                }
            }

            if (!isCorrect)
            {
                IncorrectGuesses++;
            }
```
nit: if is unecessary here and doubles the iterations

Reply:
👍🏽